app/src/main/java/com/zeronfinity/cpfy/MainActivity.kt

```diff
@@ -1,11 +1,78 @@
 package com.zeronfinity.cpfy
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.coroutines.*
+import org.json.JSONObject
+import java.net.HttpURLConnection
+import java.net.URL
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+import kotlin.coroutines.CoroutineContext
+
+class MainActivity: AppCompatActivity(), CoroutineScope {
+    lateinit var job: Job
+
+    override val coroutineContext: CoroutineContext
+        get() = job + Dispatchers.Main
+
+    val contestApiUrl = "https://clist.by/api/v1/json/contest/"
```

Comment:
You can save the url "https://clist.by/api/v1/" in a global variable.
like `ApiUrl = "https://clist.by/api/v1/"`
You may need different api url in different cases.
then you can use it as 
`ApiUrl + '/contest/'` or 
`ApiUrl + '/contest/1/'` or 
`ApiUrl + '/resource/'` or 
`ApiUrl + '/resource/1/'`
This will reduce some characters of code.

But if this is the only API call in this app then it's not necessary